braid/_braid.c

```diff
@@ -2546,12 +2579,14 @@ _braid_PeriodicCheckNumPoints(braid_Core   core,
                               braid_Int   *divisor_ptr,
                               braid_Int   *nextra_ptr)
 {
-   braid_Int  divisor, nextra, level, cfactor;
+   braid_Int  divisor, nextra, level, cfactor, computedmax;
 
    /* Compute coarsening factor divisor */
    divisor = 1;
    nextra  = 0;
-   for (level = 0; level < (_braid_CoreElt(core, max_levels) - 1); level++)
+
+   _braid_PeriodicNumLevels(core, npoints, &computedmax);
+   for (level = 0; level < computedmax; level++)
    {
       _braid_GetCFactor(core, level, &cfactor);
       if ((int)(npoints/divisor) < cfactor)
```

Comment:
@rfalgout Do you mean an additional check here like
&& (int) (npoints/divisor) < min_coarse
break;